lib/course/data.dart

```diff
@@ -151,6 +171,18 @@ class Lesson implements Entity<Lesson>, Comparable<Lesson> {
   int compareTo(Lesson other) => position.compareTo(other.position);
 
   String get webUrl => '${courseId.webUrl}/topics/$id';
+
+  @override
+  bool operator ==(Object other) =>
+      other is Lesson &&
+      id == other.id &&
+      courseId == other.courseId &&
+      name == other.name &&
+      contents.deeplyEquals(other.contents, unordered: true) &&
+      isHidden == other.isHidden &&
+      position == other.position;
+  @override
+  int get hashCode => hashList([id, courseId, name, isHidden, position]);
```

Comment:
Iterables are a bit more complicated than it seems:
Of course, `[1,2].hashCode != [1,2].hashCode` (most of the time), because two lists with the same elements aren't considered equal (`==`).
Using `dart:ui`'s `hashList` only works for ordered comparisons: `hashList([1,2]) == hashList([1,2])`, not for unordered ones like this one: `hashList([1,2]) != hashList([2,1])`.
Sorting the list items with a custom comparator that uses the `hashCode` could work. But in many performance critical sections, `hashCode` is assumed to be O(1), so I'm not sure we should do that.
That's why here I consciously opted for omitting `contents` from the `hashCode`. That means, two `Lesson`s that only differ from each other by having different contents need to be compared manually with `==` to discover they're actually not equal. I think that's acceptable.